packages/web3-providers-ipc/src/index.ts

```diff
@@ -52,19 +51,33 @@ export default class IpcProvider<API extends Web3APISpec = EthExecutionAPI> exte
 		}
 		return this._connectionStatus;
 	}
+
 	public connect(): void {
 		if (!existsSync(this._socketPath)) {
+			if (this.isReconnecting) {
+				setImmediate(() => {
+					this._reconnect();
+				});
+				return;
+			}
 			throw new InvalidClientError(this._socketPath);
 		}
-		if (!this._socketConnection || this.getStatus() === 'disconnected') {
-			this._socketConnection = new Socket();
-		}
 		try {
+			if (!this._socketConnection || this.getStatus() === 'disconnected') {
+				this._socketConnection = new Socket();
+			}
 			this._connectionStatus = 'connecting';
 			this._addSocketListeners();
 			this._socketConnection.connect({ path: this._socketPath });
 		} catch (e) {
-			throw new InvalidConnectionError(this._socketPath);
+			if (this.isReconnecting) {
+				setImmediate(() => {
+					this._reconnect();
+				});
+			} else {
+				this._connectionStatus = 'disconnected';
+				throw new InvalidConnectionError(this._socketPath);
```

Comment:
I think we should also include more error details coming via `e` in `} catch (e) {` to `throw new InvalidConnectionError(this._socketPath)`